### ISSUE TITLE
Stop shading rascal-lsp 

### ISSUE THREAD

**Issue by DavyLandman** [usethesource/rascal-language-servers#222]
**Is your feature request related to a problem? Please describe.**
We should stop using shading for the rascal-lsp.jar. It's both incorrect for what we publish to the maven repo (as the pom.xml still contains all the dependencies) .

**Describe the solution you'd like**
Avoid shading, just copy the dependencies to the rascal-vscode-extension folder

**Comment by jurgenvinju** [comment]
Is it thinkable that we load the dependencies from the .M2 folder? What are the issues there?

**Comment by DavyLandman** [comment]
Okay, we should separate this discussion in three parts:

1. what we publish to our maven repo on `https://releases.usethesource.io/maven/`
2. how we start the vscode lsp server
3. how we start the vscode repl

I think for 1, we should really stop publishing shaded jars. This strange classpaths for any down stream consumers (which might load incompatible dependencies not realizing they were already shaded inside the rascal-lsp.jar or rascal.jar). Sidenote: I just checked, at least the correct poms are published now, the shaded dependencies are removed from the list.

For 2 & 3, I'm not sure we want to enforce maven installations for our end-users, just to have a working rascal extension. Although if we manage to finish that branch of rascal with maven inlined, it might be more acceptable. It would also make starting a repl easier (assuming the correct pom.xml for a project).  It could also reduce the initial install size for the vscode extension, as we don't have to bundle the jars, we can just let maven pick it up from the usethesource repo. Downside: less self contained releases.

**Comment by jurgenvinju** [comment]
Agreed with the above. I am also interested to find out first about legal consequences. If end users are not in the know about mvn downloading stuff in the background _after_ the downloaded our core first, then what is our responsibility for the security and integrity of these downloads?

**Comment by DavyLandman** [comment]
Note, we could stop shading the rascal-lsp jar, and add all dependencies to the `jars/` asset folder. we don't have to go some complex route of downloading dependencies via maven just to get rid of shading. 

Or alternativly, we could do this in two phases. First stop shading, later switch to using maven to download all java dependencies (if we think that's a smart thing to do).

**Comment by jurgenvinju** [comment]
yes. if we stop sharing rascal-lsp.jar but not rascal.jar then we have gained a lot already.

**Comment by jurgenvinju** [comment]
I don't even know why we started shading rascal-lsp in the first place. Forgot about it I guess.

**Comment by DavyLandman** [comment]
> I don't even know why we started shading rascal-lsp in the first place. Forgot about it I guess.

it's so that rascal-vscode-extension only has to have a classpath with 3 entries. rascal, rascal-lsp and rascal-core.

if we stop shading rascal-lsp.jar, we'll get huge classpaths. we might even have to do the trick where we load the classpath from a temporary file.
